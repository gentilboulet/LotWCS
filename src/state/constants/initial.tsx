export const INITIAL_STATE = 'INITIAL_STATE';
export type INITIAL_STATE = typeof INITIAL_STATE;
