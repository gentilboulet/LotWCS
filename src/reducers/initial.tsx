import { makeTypedFactory } from 'typed-immutable-record';
import * as Immutable from 'immutable';
import { IStoreStateJS, IStoreState,
  IStoreSkill, skillFactory,
  IStoreSkillSpecialityJS,
  IStoreLoresheet } from '../types/state';

import { resetToInitialState } from '../actions/initial';
import { IAction } from '../types/actions';
import { IDataSkill } from '../types/skills';
import { skills } from '../data/skills';

export const defaultStateJS: IStoreStateJS = {
  name: 'No Name',
  concept: 'No Concept',
  archetype: '',
  archetypeModified: false,
  rank: '',
  rankValue: -1,
  chi: 0,
  rankModified: false,
  entanglement: 0,
  destiny: 0,
  skills: Immutable.List<IStoreSkill>(skills.map((s: IDataSkill) => {
    return skillFactory({
      name: s.name,
      value: 0,
      specialities: Immutable.List<IStoreSkillSpecialityJS>(s.specialities.map(
        (spe: string) => { return {name: spe, bought: false}; }
      )),
    });
  })),

  loresheets: Immutable.List<IStoreLoresheet>([]),
  reductions: Immutable.List([]),
  bonuses: Immutable.List([]),

  history: Immutable.List<IAction>([resetToInitialState()]),
};

export const initialStateFactory = makeTypedFactory<IStoreStateJS, IStoreState>(defaultStateJS);
