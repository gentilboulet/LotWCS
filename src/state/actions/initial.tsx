import * as constants from 'constants/initial';

export interface IInitialState {
  type: constants.INITIAL_STATE;
}

export type IInitialStateAction = IInitialState;

export function resetToInitialState(): IInitialState {
  return {type: constants.INITIAL_STATE};
}
