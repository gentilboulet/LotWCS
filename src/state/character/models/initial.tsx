/* eslint-disable import/first */
import { createState as createAutomaticsState } from "./automatics";
import { createState as createChiState } from "./chi";
import { createState as createDiscountsState } from "./discounts";
import { createState as createaKungFuState } from "./kungfu";
import { createState as createLoresheetsState } from "./loresheets";
import { createState as createSkillsState } from "./skills";
import { ICharacterState } from "./type";
import { createState as createVirtuesState } from "./virtues";

export function emptyStateFactory(): ICharacterState {
  return {
    archetype: undefined,
    concept: undefined,
    name: undefined,
    rank: undefined,

    destiny: 0,
    entanglement: 0,

    chi: createChiState(),
    kungfu: createaKungFuState(),
    loresheets: createLoresheetsState(),
    skills: createSkillsState(),
    virtues: createVirtuesState(),

    automatics: createAutomaticsState(),
    discounts: createDiscountsState()
  };
}

import { setArchetype, setRank } from "../actions/header";
import { skillSpecialityBuy } from "../actions/skills";
import { playActions } from "../reducers/global";
import { zeroCost } from "./costs";

export function testingStateFactory(): ICharacterState {
  const initial = emptyStateFactory();
  const actions = [
    setRank(2),
    setArchetype("warrior"),
    skillSpecialityBuy("Awareness", "Hear", zeroCost),
    skillSpecialityBuy("Awareness", "Sight", zeroCost)
  ];
  return playActions(initial, actions);
}

export function initialStateFactory(): ICharacterState {
  return emptyStateFactory();
}
