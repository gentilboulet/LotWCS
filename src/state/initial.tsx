import { IStoreState } from "./type";

import { createState as createHistoryState } from "./history";

import { createState as createChiState } from "./chi";
import { createState as createaKungFuState } from "./kungfu";
import { createState as createLoresheetsState } from "./loresheets";
import { createState as createSkillsState } from "./skills";
import { createState as createVirtuesState } from "./virtues";

import { createState as createBonusesState } from "./bonuses";
import { createState as createDiscountsState } from "./discounts";

function emptyInitialStateFactory(): IStoreState {
  return {
    archetype: undefined,
    concept: undefined,
    name: undefined,
    rank: undefined,

    destiny: 0,
    entanglement: 0,

    history: createHistoryState(),

    chi: createChiState(),
    kungfu: createaKungFuState(),
    loresheets: createLoresheetsState(),
    skills: createSkillsState(),
    virtues: createVirtuesState(),

    bonuses: createBonusesState(),
    discounts: createDiscountsState()
  };
}

import { setArchetype, setRank } from "./actions/header";
import { skillSpecialityBuy } from "./actions/skills";
import { zeroCost } from "./costs";
import { replayHistory } from "./history";

export function testingStateFactory(): IStoreState {
  const initial = emptyInitialStateFactory();
  const actions = [
    setRank("4th_rank"),
    setArchetype("warrior"),
    skillSpecialityBuy("Awareness", "Hear", zeroCost),
    skillSpecialityBuy("Awareness", "Sight", zeroCost)
  ];
  initial.destiny += 5;
  return replayHistory(initial, actions);
}

export function initialStateFactory(): IStoreState {
  return emptyInitialStateFactory();
}
