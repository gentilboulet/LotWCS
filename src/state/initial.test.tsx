import { initialStateFactory } from './initial';

describe('Testing initialStateFactory', () => {
  it('should be constant', () => {
    expect( initialStateFactory() ).toMatchSnapshot();
  });
});
