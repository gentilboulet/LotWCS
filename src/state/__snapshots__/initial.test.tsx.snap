// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing initialStateFactory should be constant 1`] = `
Object {
  "archetype": undefined,
  "bonuses": Array [],
  "chi": Object {
    "corrupt": Object {
      "cultivation": 0,
      "value": 0,
    },
    "earth": Object {
      "cultivation": 0,
      "value": 0,
    },
    "enlightened": Object {
      "cultivation": 0,
      "value": 0,
    },
    "fire": Object {
      "cultivation": 0,
      "value": 0,
    },
    "general": Object {
      "cultivation": 0,
      "value": 0,
    },
    "metal": Object {
      "cultivation": 0,
      "value": 0,
    },
    "water": Object {
      "cultivation": 0,
      "value": 0,
    },
    "wood": Object {
      "cultivation": 0,
      "value": 0,
    },
  },
  "concept": undefined,
  "destiny": 0,
  "discounts": Array [],
  "entanglement": 0,
  "history": Array [
    Object {
      "type": "INITIAL_STATE",
    },
  ],
  "kungfu": Object {
    "KUNGFU_EXTERNAL": Object {},
    "KUNGFU_INTERNAL": Object {},
  },
  "loresheets": Object {},
  "name": undefined,
  "rank": undefined,
  "skills": Object {},
  "virtues": Array [
    Object {
      "name": "Honor",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Benevolence",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Righteousness",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Loyalty",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Force",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Revenge",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Individualism",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Obsession",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Ruthlessness",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Ferocity",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
  ],
}
`;
